src/main/java/com/minecolonies/coremod/colony/requestsystem/management/handlers/RequestHandler.java

```diff
@@ -509,38 +522,41 @@ public static IRequest getRequest(final IStandardRequestManager manager, final I
      * @param token The token to get the request for.
      * @return The request or null when no request with that token exists.
      */
-    public static IRequest getRequestOrNull(final IStandardRequestManager manager, final IToken token)
+    @SuppressWarnings(RAWTYPES)
+    public static IRequest getRequestOrNull(final IStandardRequestManager manager, final IToken<?> token)
     {
         LogHandler.log("Retrieving the request for: " + token);
 
-        return manager.getRequestBiMap().get(token);
+        return manager.getRequestIdentitiesDataStore().getIdentities().get(token);
     }
 
     /**
      * Wrapper for a assignment result.
      */
-    private final static class AssigningResult implements Comparable<AssigningResult>
+    private static final class AssigningResult<T> implements Comparable<AssigningResult<T>>
     {
-        private final IRequestResolver resolver;
-        private final List<IToken>     children;
+        @SuppressWarnings(RAWTYPES) private final IRequestResolver resolver;
+        private final List<IToken<T>>  children;
 
-        private AssigningResult(final IRequestResolver resolver, final List<IToken> children)
+        @SuppressWarnings(RAWTYPES)
+        private AssigningResult(final IRequestResolver resolver, final List<IToken<T>> children)
         {
             this.resolver = resolver;
-            this.children = children;
+            this.children = new ArrayList<>(children);
         }
 
-        public List<IToken> getChildren()
+        public List<IToken<T>> getChildren()
         {
-            return children;
+            return Collections.unmodifiableList(children);
         }
 
         @Override
-        public int compareTo(@NotNull final AssigningResult o)
+        public int compareTo(@NotNull final AssigningResult<T> o)
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Override "equals(Object obj)" to comply with the contract of the "compareTo(T o)" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1210)